docs/user/agent/authentication-authorization.md

```diff
@@ -0,0 +1,124 @@
+
+For example:  
+- Suppose the endpoint `/api/assisted-install/v2/infra-envs` requires the `watcherAuth` scheme.  
+- A request is made with the following command, but an **incorrect token header** is sent:  
+  ```bash
+  curl http://localhost:8090/api/assisted-install/v2/infra-envs \
+       -H "Authorization: eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdXRoX3NjaGVtZSI6IndhdGNoZXJBdXRoIn0.hJ5EuBOttDQ1-ANWUJlc97H1k63d-clDCaFUZ84XSYdF03A938OfMU6RHWltv0AiWZEHNJ0gLrjLxR-1yUYqiA"
+  ```  
+- Although the token specifies the correct `auth_scheme` (`watcherAuth`), the **header sent is incorrect** for this request.
+
+Expected response:  
+```json
+{
+  "code": 403,
+  "message": "authClaim watcherAuth is unauthorized to access"
+}
+```
+
+**Outcome:** 
+The system identifies that the provided header does not satisfy the endpoint’s requirements and correctly responds with an error. This confirms that tokens must not only align with the expected `authScheme` but also be passed in the correct format and header. The authorization logic ensures that only authorized requests can access protected endpoints in Assisted Service.
+
+## Authentication API (Server Side)
+
+The authentication server validates the token provided by the client using the `validateToken` function. The following steps explain the server-side validation process:
+
+1. When an agent or nodes ISO is created, a set of public and private keys is generated. The private key is used to sign the JWT tokens, while the associated public key is made available to the API server.
+2. The token is passed to the API server as part of the request. The API server validates the token by verifying its signature against the public key. This ensures the token was signed by the correct private key and has not been tampered with.
+3. If the token is invalid or expired, an error is logged and the request is rejected with a `Unauthorized` error.
+4. The claims in the token are extracted.
+5. If the `exp` (expiration) claim is found, the server checks whether the token has expired.
+   - In the install workflow, the `exp` claim will not be present.
+   - In the add-nodes workflow, the token expiration is checked. If expired, a new token must be generated.
+6. If the token is valid and not expired, the claims are returned and the request proceeds.
+
+## Expired Token Handling on Day 2
+
+When a node is booted using the Day 2 `nodes.iso` to add worker nodes, the `agent-auth-token-status.service` systemd service will check if the authentication tokens have expired. If the tokens are expired, the following message will be displayed:
+
+```
+The authentication token has expired. Please generate a new ISO using the "oc adm node-image create" command, then reboot the node.
+```
+
+This ensures that the user is notified of expired tokens and can take appropriate action to generate a new ISO.
+
+## Conclusion
```

Comment:
Is that really needed? It seems to me that the motivations have been clearly described in the introduction paragraph